felix/wireguard/wireguard.go

```diff
@@ -1008,25 +1045,25 @@ func (w *Wireguard) updateCacheFromNodeUpdates(conflictingKeys set.Set) {
 			delete(w.nodeUpdates, name)
 		}
 	}
+
+	return conflictingKeys
 }
 
 // updateRouteTable updates the route table from the node updates.
 func (w *Wireguard) updateRouteTableFromNodeUpdates() {
-	// Do all deletes first. Then adds or updates separarately. This ensures a CIDR that has been deleted from one node
+	// Do all deletes first. Then adds or updates separately. This ensures a CIDR that has been deleted from one node
 	// and added to another will not add first then delete (which will remove the route, since the route table does not
 	// care about destination node).
-	for name, update := range w.nodeUpdates {
-		// Delete routes that are no longer required in routing.
-		node := w.getOrInitNodeData(name)
-		ifaceName := routetable.InterfaceNone
-		if node != nil && node.programmedInWireguard {
-			ifaceName = w.config.InterfaceName
-		}
-		logCxt := log.WithFields(log.Fields{"node": name, "ifaceName": ifaceName})
+	for _, update := range w.nodeUpdates {
+		// Delete routes that are no longer required in routing. Just delete both the wireguard and throw routes - this
+		// is somewhat defensive as we havw the information to decide which route we need to remove - however we have
```

Comment:
```suggestion
		// is somewhat defensive as we have the information to decide which route we need to remove - however we have
```

Reply:
What about the one below on line 1117, worth doing the unconditional delete there as well?

Reply:
darn it .... coulda sworn i'd looked for that!  Thanks!  Done.